src/app/components/DocumentView/DocumentView.tsx

```diff
@@ -1,22 +1,74 @@
-import React from 'react';
+import React, { useCallback } from 'react';
+import twoColumnDocument from '../../mocks/2-column-document.json';
+import twoPageDocument from '../../mocks/2-page-document.json';
+import { MockDocumentDataType } from '../../types/document.type';
+import { generateHtml } from '../../utils/helper';
+import PDFConverter from '../PdfConverter/PdfConverter';
+import './documentStyles.css';
 
 const DocumentView = () => {
-  const onGenerate = () => {
-    fetch('/api/generate', {
-      method: 'GET',
-    });
-  };
+  const [isPdfGenerated, setIsPdfGenerated] = React.useState(false);
+  const [htmlData, setHtmlData] = React.useState<string>('');
+  const [imageUri, setImageUri] = React.useState<string | ArrayBuffer | null>(
+    ''
+  );
+  const [documentName, setDocumentName] = React.useState<string>('');
+  const onGenerate = useCallback(async (document: MockDocumentDataType) => {
+    const { finalHtmlData, imageUri, documentName } = await generateHtml(
```

Comment:
out of curiosity, did you consider doing this server side? i.e. hit an API that starts generating the HTML and returns the html / triggers a download?
(Not sure how jsPDF interacts or whether it has to be client side)